themes/digital.gov/src/js/guide-sidenav.js

```diff
@@ -1,18 +1,106 @@
-document.addEventListener("DOMContentLoaded", () => {
+// Return all of the h2 and h3 elements in the guide content body
+function getHeadings() {
   const guideNav = document.querySelector(".dg-guide__nav-list");
+  if (guideNav) {
+    const body = document.querySelector(".dg-guide__content-body");
+    if (body) {
+      const headings = body.querySelectorAll(
+        "h2:not(.dg-featured-resource__text-title), h3:not(.usa-accordion__heading)"
+      );
+      return headings;
+    }
+  }
+  return [];
+}
 
+function scrollIntoView(element) {
+  document.querySelector(element).scrollIntoView({
+    behavior: "smooth",
+  });
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+  const guideNav = document.querySelector(".dg-guide__nav-list");
   if (!guideNav) return;
 
-  // add class to current sidenav list item
+  // Add "read" class to previously read sections
   const guideCurrentListItem = guideNav.querySelector(".usa-current");
-  guideCurrentListItem.parentNode.classList.add("current");
-
-  // apply style to previous list item to display guide progress
+  if (guideCurrentListItem) {
+    guideCurrentListItem.parentNode.classList.add("current");
+  }
   const sidenavLinks = document.querySelectorAll(".usa-sidenav__item");
 
   // eslint-disable-next-line no-restricted-syntax
   for (const link of sidenavLinks) {
-    if (link.classList.contains("current")) return;
+    if (link.classList.contains("current")) break;
     link.classList.add("usa-sidenav__item--read");
   }
+
+  // Add headings as subnav to current section
+  const headings = getHeadings();
+  const subList = document.createElement("ul");
+  subList.classList.add("usa-sidenav__sublist");
+
+  for (let i = 0; i < headings.length; i += 1) {
+    // Clean heading text and create link to append to subnav
+    const text = headings[i].innerText;
+    const regex = /[!"#$%&'()*+,./:;<=>?@[\]^_`{|}~“”‘’]/g;
+    const cleanText = text.replace(regex, "");
+    const textElements = cleanText.split(" ");
+    const href = textElements.join("-").toLowerCase();
+    const link = document.createElement("a");
+    link.href = `#${href}`;
+    link.innerText = text;
+    link.addEventListener("click", (e) => {
+      e.preventDefault();
+      scrollIntoView(link.getAttribute("href"));
+
+      // This is necessary to account for a bug with multiple scroll listeners in Chromium browsers
+      const menuBar = document.querySelector(".dg-guide__menu-bar");
```

Comment:
What's this functionality for?

It seems odd that we have code for the menu bar (this is the top-level menu) in the sidenav.

Reply:
This is part of the Chromium bug. Basically, the menu bar automatically scrolls itself when it sticks to the top of the page, which interrupts the scroll from the in page navigation. As a result, we need to know if the menu bar is stuck to implement the work around...

Reply:
Is there an issue to the Chromium bug so we can track/update when it gets resolved?

Reply:
Yeah I have been tracking [this issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1043933) but I believe there are other ones related to it (like [this one](https://bugs.chromium.org/p/chromium/issues/detail?id=1035302))